core/service/storage-service.js

```diff
@@ -108,42 +108,53 @@ var StorageService = exports.StorageService = Montage.specialize({
-                    }).then(function(newVolblocksize) {
-                        newVolblocksize.parsed = 512;
-                        dataset.properties.volblocksize = newVolblocksize;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefreservation);
-                    }).then(function(newRefreservation) {
-                        dataset.properties.refreservation = newRefreservation;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyReservation);
-                    }).then(function(newReservation) {
-                        dataset.properties.reservation = newReservation;
+                    promises.push(
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyAtime).then(function(newAtime) {
+                            dataset.properties.atime = newAtime;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyCasesensitivity).then(function(newCasesensitivity) {
+                            dataset.properties.casesensitivity = newCasesensitivity;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyCompression).then(function(newCompression) {
+                            dataset.properties.compression = newCompression;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyDedup).then(function(newDedup) {
+                            dataset.properties.dedup = newDedup;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyQuota).then(function(newQuota) {
+                            dataset.properties.quota = newQuota;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefquota).then(function(newRefquota) {
+                            dataset.properties.refquota = newRefquota;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyVolblocksize).then(function(newVolblocksize) {
+                            dataset.properties.volblocksize = newVolblocksize;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefreservation).then(function(newRefreservation) {
+                            dataset.properties.refreservation = newRefreservation;
+                        }),
+                        self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyReservation).then(function(newReservation) {
+                            dataset.properties.reservation = newReservation;
+                        })
+                    );
+
+                    return Promise.all(promises).then(function(){
```

Comment:
Doing initialization in two steps (the first one adds the property to `properties`, the second one put default values in the property) make execution flow harder to follow, it may be better to have a promise that resolves once the property is fully initialized.